CRM/Utils/Geocode/Google.php

```diff
@@ -82,7 +82,7 @@ public static function format(&$values, $stateName = FALSE) {
       $add .= ',+';
     }
 
-    if (!empty($values['state_province']) || !empty($values['state_province_id'])) {
+    if (!empty($values['state_province']) || (!empty($values['state_province_id']) && $values['state_province_id'] != 'null')) {
```

Comment:
Reason for not using isNull - (we don't have isEmpty() :-(), is what I thought to check before, but that function only works if the key is actually present in the array and throws notices when it is not. Eg:

    $var = array('key' => 'null');
    if (!CRM_Utils_System::isNull($var['key'])) {
      CRM_Core_Error::Debug('Not Null', $var);
    }

checks correctly and all seems to be working fine. Now, if we don't get this key:

    $var = array();
    if (!CRM_Utils_System::isNull($var['key'])) {
      CRM_Core_Error::Debug('Not Null', $var);
    }

will check for the condition correctly but will also throw a notice `Notice: Undefined index: key` since the key is not present and check of isset() is done after param is entered into isNull() function.

'null' value is being set at this point https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/Address.php#L319  to fix CRM-3393. Also, it is a very minor change needed, hence proceeded with a simple check other than manipulating 'null' to an empty string and check for all remaining scenarios.